packages/components/src/text-input/hooks/use-text-input-state-utils.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import { useDrag } from 'react-use-gesture';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+
+import { clearSelection } from '../../utils/clear-selection';
+
+import * as styles from '../styles';
+
+/**
+ * @typedef Props
+ * @property {(int: number) => void} [increment] Increment text input number value callback.
+ * @property {(int: number) => void} [decrement] Decrement text input number value callback.
+ * @property {boolean} [isTypeNumeric] Whether the type is numeric.
+ * @property {string} [dragAxis] The drag axis, e.g., `x` or `y`.
+ */
+
+/**
+ * @param {Props} props
+ * @return {Object} gestures
```

Comment:
If we remove the return type TypeScript should correctly infer the return type without us needing to specify it. Just need to delete this line.